brave/brave-core
# fix(wallet): Edit Visible Assets Modal Token List

## Description 
- Fixes a bug where the `Edit Visible Assets` modal was not returning the `Full Token List` on second open.
- Fixes a bug where removing a token from the `Edit Visible Assets` modal would not update in the `Portfolio` unless you refreshed the page.
- Performance improvement: Removed the `useTokenRegistry` hook and now utilizes the `useGetTokensRegistryQuery` cached data.


Resolves <https://github.com/brave/brave-browser/issues/31051>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

   1. Open the `Wallet` and open the `Edit Visible Assets` modal
   2. Scroll the list, it shows the `Full token list`
   3. Close the `Edit Visible Assets` modal and reopen it
   4. Scroll the list, it should still show the `Full token list`

Before:

https://github.com/brave/brave-core/assets/40611140/73ba79e6-0157-418c-b355-186f77237935

After:

https://github.com/brave/brave-core/assets/40611140/11fb2e00-84ed-4205-a779-b724b0526253
